crates/db_views/src/post_view.rs

```diff
@@ -350,6 +319,35 @@ impl<'a> PostQuery<'a> {
         .then_order_by(post_aggregates::featured_community.desc());
     }
 
+    if let Some(creator_id) = self.creator_id {
+      query = query.filter(post::creator_id.eq(creator_id));
+    }
+
+    if self.community_id.is_none() && self.creator_id.is_none() {
```

Comment:
I added this `community_id.is_none() && creator_id.is_none()` check. Even if `ListingType` ends up being `Some`, its best to ignore it if you're looking at either a profile page, or a specific community.

Reply:
This will probably look like a bug to someone using the API. Isnt the better fix here not to send these IDs from lemmy-ui if they arent needed?

Reply:
`GetPersonDetails` [doesn't send the listing_type](https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/api/read_person.rs#L69), and its not a part of the API call, so I spose I could remove this then.